archetype-common/src/main/java/org/apache/maven/archetype/creator/FilesetArchetypeCreator.java

```diff
@@ -1632,6 +1634,18 @@ private FileSet getUnpackagedFileSet(
         return createFileSet(excludes, false, filtered, group, includes, defaultEncoding);
     }
 
+    private String getFileCharsetEncoding(File detectedFile, String defaultEncoding) {
+        try (FileInputStream fileInputStream = new FileInputStream(detectedFile);
+                BufferedInputStream is = new BufferedInputStream(fileInputStream)) {
```

Comment:
put both streams into the try block argument

Reply:
use finally to close these streams?

Reply:
Sorry, I don't understand what you mean actually. But I've changed my code, please take a look.

Reply:
I filed the issue because the helper class was very poorly designed and implemented. I didn't initially realize it was only used in one place and could be inlined so neatly and trivially. However it turns out that it can, and this PR noticeably simplifies the archetype creator class as well as eliminating the charset detectpr class. It's good work.